qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -0,0 +1,51 @@
+from tornado.web import authenticated, HTTPError
+
+from qiita_pet.handlers.base_handlers import BaseHandler
+from qiita_pet.util import is_localhost
+from qiita_pet.handlers.util import download_link_or_path
+from qiita_db.util import get_files_from_uploads_folders
+from qiita_pet.handlers.api_proxy import (
+    sample_template_info, process_sample_template, update_sample_template,
+    delete_sample_template, get_sample_template_filepaths,
+    study_data_types_proxy)
+
+
+class SampleTemplateAJAX(BaseHandler):
+    @authenticated
+    def get(self):
+        """Send formatted summary page of sample template"""
+        study_id = self.get_argument('study_id')
+        files = [f for _, f in get_files_from_uploads_folders(study_id)
+                 if f.endswith(('txt', 'tsv'))]
+        data_types = sorted(study_data_types_proxy())
+        is_local = is_localhost(self.request.headers['host'])
+        # Get the most recent version for download and build the link
+        download = get_sample_template_filepaths(study_id,
+                                                 self.current_user.id)[-1]
+        dl_path = download_link_or_path(
+            is_local, download[0], download[1], "Download sample information")
+
+        stats = sample_template_info(study_id, self.current_user.id)
+        self.render('study_ajax/sample_summary.html', stats=stats['summary'],
+                    num_samples=stats['num_samples'], dl_path=dl_path,
+                    files=files, study_id=study_id, data_types=data_types)
+
+    @authenticated
+    def post(self):
+        """Edit/delete/recreate sample template"""
+        action = self.get_argument('action')
```

Comment:
Yes. The quote UI (which is this handler) calls the rest interface to do everything. Sine we can only get and post through web forms, this is a needed check.

> On Dec 22, 2015, at 09:30, Daniel McDonald <notifications@github.com> wrote:
> 
> In qiita_pet/handlers/study_handlers/sample_template.py <https://github.com/biocore/qiita/pull/1582#discussion_r48276731>:
> 
> > +        is_local = is_localhost(self.request.headers['host'])
> > +        # Get the most recent version for download and build the link
> > +        download = get_sample_template_filepaths(study_id,
> > +                                                 self.current_user.id)[-1]
> > +        dl_path = download_link_or_path(
> > +            is_local, download[0], download[1], "Download sample information")
> > +
> > +        stats = sample_template_info(study_id, self.current_user.id)
> > +        self.render('study_ajax/sample_summary.html', stats=stats['summary'],
> > +                    num_samples=stats['num_samples'], dl_path=dl_path,
> > +                    files=files, study_id=study_id, data_types=data_types)
> > +
> > +    @authenticated
> > +    def post(self):
> > +        """Edit/delete/recreate sample template"""
> > +        action = self.get_argument('action')
> Isn't this a hybrid SOAP + REST?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/biocore/qiita/pull/1582/files#r48276731>.
>